trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/credentials/StsController.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.credentials;
+
+import java.util.Optional;
+
+public interface StsController
+{
+    Optional<StsAssumedRole> assumeRole(
+            SigningMetadata signingMetadata,
+            String requestArn,
+            Optional<String> requestExternalId,
+            Optional<String> requestRoleSessionName,
+            Optional<Integer> requestDurationSeconds);
+
+    default Optional<Credentials> checkAssumedRole(Optional<String> session)
```

Comment:
From the PR comment:
> When the S3 Proxy endpoint receives a request. It first checks
the StsController to see if the emulated AccessKey and Session
map to an assumed role.

However, this is currently only checking the existence of an assumed role based on the session token. Given that STS would return a new set of `(accessKey, secretAccessKey, sessionToken)` values, the interface should at least verify the existence of a session based on an `accessKey` and `sessionToken` pair right?

The `sessionToken` format is not documented on the AWS side, so I can imagine some people may want to use it for things like issuing an expiry date (as a signed JWT or something), so realistically there could be clashes I think

Reply:
So, add the accessKey? Sure, makes sense.